### DISCUSSION TITLE
Feedback: listen()

### DISCUSSION THREAD

**Discussion by Pho3nixHun** [mswjs/mswjs.io#390] [Q&A] [answered]
- [Open page](https://mswjs.io/docs/api/setup-server/listen/)

## Feedback

The documentation treats the url property as a URL,
```ts
server.listen({
  onUnhandledRequest(request, print) {
    // Ignore requests to fetch static assets.
    if (request.url.pathname.includes('/assets/')) {
      return
    }
 
    // Otherwise, print a warning for any unhandled request.
    print.warning()
  },
})
```
 while in reality it is just a string. You have to manually convert it to an url:
```ts
const url = new URL(request.url)
if (url.pathname.includes('/assets/')) {
...
}
```

**Comment by kettanaito** [comment]
Hi, @Pho3nixHun. That's a great find! It's a leftover from the v1 docs where `request.url` was indeed a URL instance. I will fix that shortly.

**Comment by kettanaito** [comment, marked as answer]
I've merged the fix, it should be live now. Once again, thanks for spotting this! If you find any other discrepancies in the docs, please let me know.
